_posts/en/meetings/2016-11-17-meeting.md

```diff
@@ -0,0 +1,146 @@
+name: 2016-11-17-meeting
+type: meetings
+layout: page
+lang: en
+version: 1
+---
+{% include _toc.html %}
+ 
+- [Link to this week logs](https://botbot.me/freenode/bitcoin-core-dev/2016-11-17/?msg=76630924&page=2)
+- [Meeting minutes by meetbot](http://www.erisian.com.au/meetbot/bitcoin-core-dev/2016/bitcoin-core-dev.2016-11-17-19.04.html)
+ 
+---
+
+## Main topics
+ 
+- Shared_ptr
+- Priority removal
+- Account removal
+- Auxiliary block requests
+
+## Shared_ptr
+
+### background
+
+In many places we've started using 'shared_ptr' instead of the object itself, this way it can be shared between many data structures without making copies of it.
+
+### meeting comments
+
+Sipa has a series of 3 pull requests to introduce shared_ptr in more places, namely [#9125][] (Make CBlock a vector of shared_ptr of CTransactions), [#8580][](Make CTransaction actually immutable) and [#8589][] (Inline CTxInWitness inside CTxIn (on top of #8580)). The first being a necessary refactor for the ones that follow and a performance improvement of 3-4%. The second may be more controversial as it affects the wallet code significantly. Wumpus thinks the old behavior of CwalletTx inheriting from CTransaction is a good example of abuse of inheritance.
+
+### meeting conclusion
+
+- Review [#9125][]
+
+## Priority removal
+
+### background
+
+The priority system used to assign transactions with a priority based on age, size, and number of inputs, which made some transactions free. This has a large codebase and efforts have been made to remove the system as it can't be expected from miners to keep including 0-fee transactions.
+
+### meeting comments
+
+Morcos notes the priority code doesn't serve any function anymore, perhaps a target can be set to remove all the priority code for 0.15. There's a lot of stuff merged on the way for priority already.
+
+Gmaxwell agrees and thinks any desire to keep priority around could be answered by intelligent use of fee delta.
+
+There's 4 components to "removing priority": rpc, estimation, mining and relay. Estimation has already been removed. Gmaxwell would like to see relay removed as it currently causes bandwidth wasting as it's relaying around transactions that won't get mined.
+
+### meeting conclusion
+
+- Change default to disable priority relay.
+- Rehash with Luke-Jr, as he's the main proponent of keeping priority but not present at meeting
+
+## Account removal
+
+### background
+
+Currently bitcoin-core uses a system of accounts. Third parties using this can have multiple problems with this, and it's been the consensus [for a long time](https://github.com/bitcoin/bitcoin/issues/3816) the system should be removed and replaced by a 'label' system.
+
+### meeting comments
+
+Earlier this year Wumpus opened a [pull request][#7729] to introduce a 'label' API for the wallet (previously mentioned in the [2016-07-14 meeting](en/meetings/2016/07/14/#notes--short-topics)). It still hasn't received much review.

```

Comment:
looks like the link is missing a leading `/` -> `/en/meetings/2016/07/14/#notes--short-topics`